File: py/Task1.py
class Solution:
    def twoSum(self, nums:[int], target: int) -> [int]:
        dict = {}
        for idx,num in enumerate(nums):
            if num in dict:
                return [dict[num],idx]
            else:
                dict[target-num] = idx

solution = Solution()
print(solution.twoSum([2, 7, 11, 15],9))

File: go/src/task1/task1.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
	set := make(map[int]int)
	for p, num := range nums {
		if q, ok := set[num]; ok {
			return []int{p, q}
		}
		set[target-num] = p
	}
	return nil
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(nums, target))
}
